task.go

```diff
@@ -186,3 +214,123 @@ func (t *task) Resize(ctx context.Context, w, h uint32) error {
 	})
 	return err
 }
+
+func WithExit(r *execution.CheckpointRequest) error {
+	r.Exit = true
+	return nil
+}
+
+func (t *task) Checkpoint(ctx context.Context, opts ...CheckpointOpts) (digest.Digest, error) {
```

Comment:
Would be better if this returned the whole descriptor rather just the digest. Knowing the media type could be useful.